### PR TITLE
Support CLE's Case Search filter in GeoSpatial Reports

### PR DESCRIPTION
**Pull request by mkangia**
## Product Description

Update reports under GEOSPATIAL to support Case Search filter like on CLE instead of the simple case list like filter.

## Technical Summary

https://dimagi-dev.atlassian.net/browse/SC-3252

Originally, the filters added by CaseListMixin were being used. So 'fields` was overwritten to change filters to have `XPathCaseSearchFilter` instead of `CaseListFilter`. And then the code to implement that filter values, was extracted out and reused in the view modified.

## Feature Flag

`GEOSPATIAL`

### Safety story

Tested locally.
The change to extract a mixin out of `CaseListExplorer` should not change anything about it.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by zandre-eng** [comment]
I see there is nothing under the QA section. @mkangia will this PR be taken through QA? (I see it was requested as a part of the ticket's requirements)

**Review by SmittieC — APPROVED** [review]
Looks good to me. There's a slight concern at the back of my head about the discoverablility of the `XpathCaseSearchFilterMixin`, but I'm not sure what the best course of action should be. I wonder if it might not be better moving it to the `filters` file, since this is a type of filter.

**Comment by mkangia** [comment]
Thanks for flagging that @zandre-eng . I am not sure why QA was expected for this. It does not seem necessary for the changes done. Is it okay for me add a comment on the ticket and skip QA?

**Comment by mkangia** [comment]
Good point @SmittieC. I just left the mixin in the view file itself since it was a view mixin.
Should I add a new file called mixin and put it under that? I see we have such a file for some apps

**Comment by zandre-eng** [comment]
> Is it okay for me add a comment on the ticket and skip QA?

@mkangia Given the scope of the changes, I agree putting this through QA isn't necessary. A comment on the ticket sounds good.

**Review by zandre-eng — APPROVED** [review]

**Comment by SmittieC** [comment]
> Good point @SmittieC. I just left the mixin in the view file itself since it was a view mixin. Should I add a new file called mixin and put it under that? I see we have such a file for some apps

@mkangia That seems fine. Thinking about it a bit more, I think my concern is this: How do I know I have to use this mixin when using the `XPathCaseSearchFilter` in a view? I think a comment or example in the docstring of the `XPathCaseSearchFilter` class might help?

**Comment by mkangia** [comment]
@SmittieC I see.
I named them so that is a hint there, mixin is called `XpathCaseSearchFilterMixin` and the filter is `XPathCaseSearchFilter`.
Adding a docstring to `XPathCaseSearchFilter` sounds good to me.

**Review by SmittieC — APPROVED** [review]

File: corehq/apps/reports/standard/cases/case_list_explorer.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def es_results(self):
     def _build_query(self, sort=True):
         query = super(CaseListExplorer, self)._build_query()
         query = self._populate_sort(query, sort)
-        query = self.apply_xpath_case_search_filter(self, query)
+        query = self.apply_xpath_case_search_filter(query)
```
Comment (mkangia) [inline corehq/apps/reports/standard/cases/case_list_explorer.py:94]: Bad miss from my end. Glad I could catch it.

Comment (SmittieC) [reply corehq/apps/reports/standard/cases/case_list_explorer.py:94]: Ah, good one

**Review by SmittieC — APPROVED** [review]

**Comment by mkangia** [comment]
Tested on staging that this is working well. Just waiting on go ahead on the ticket for not going through QA for this

**Pull request merged**